src/par_iter/internal.rs

```diff
@@ -7,14 +7,20 @@ use join;
 use super::IndexedParallelIterator;
 use super::len::*;
 
+pub trait ProducerContinuation<ITEM> {
+    type Output;
+    fn with_producer<'p, P>(self, producer: P, shared: &'p P::Shared) -> Self::Output
+        where P: Producer<'p, Item=ITEM>;
+}
+
```

Comment:
No, you would have to do so. The idea of the `internal` module was that this is code you only need *if* you are writing custom parallel iterators. Maybe it needs a better name.